tests_consume/test_via_engine_api.py

```diff
@@ -1,25 +1,197 @@
+) -> Mapping[str, io.BufferedReader]:
+    """
+    Defines the files that hive will start the client with.
+    The buffered genesis including the pre-alloc.
+    """
+    files = {}
+    # Client specific genesis format
+    if client_type.name == "nethermind":
+        files["/chainspec/test.json"] = buffered_genesis
+    else:
+        files["/genesis.json"] = buffered_genesis
+    return files
+
+
+@pytest.fixture(scope="function")
+def client_environment(test_case: TestCase) -> Dict:
+    """
+    Defines the environment that hive will start the client with
+    using the fork rules specific for the simulator.
+    """
+    client_env = {
+        "HIVE_FORK_DAO_VOTE": "1",
+        "HIVE_CHAIN_ID": "1",
+        "HIVE_NODETYPE": "full",
+        **{k: f"{v:d}" for k, v in client_fork_ruleset.get(test_case.fixture.fork, {}).items()},
+    }
+    return client_env
 
 
 @pytest.fixture(scope="function")
-def client(hive_test: HiveTest, files: dict, environment: dict, client_type: ClientType) -> Client:
+def client(
+    hive_test: HiveTest, client_files: Dict, client_environment: Dict, client_type: ClientType
+) -> Client:
     """
-    Return the hive client being used to execute the current test case.
+    Initializes the client with the appropriate files and environment variables.
     """
-    client = hive_test.start_client(client_type=client_type, environment=environment, files=files)
+    client = hive_test.start_client(
+        client_type=client_type, environment=client_environment, files=client_files
+    )
     assert client is not None
     yield client
     client.stop()
 
 
-@pytest.mark.skip(reason="Engine API consumer not implemented yet.")
-def test_via_engine_api(test_case: TestCase, client: Client):  # noqa: D103
-    assert test_case is not None
+@pytest.fixture(scope="function")
+def engine_rpc(client: Client) -> EngineRPC:
+    """
+    Initializes the engine RPC for the client under test.
+    """
+    return EngineRPC(client)
+
+
+@pytest.fixture(scope="function")
+def eth_rpc(client: Client) -> EngineRPC:
+    """
+    Initializes the eth RPC for the client under test.
+    """
+    return EthRPC(client)
+
+
+@pytest.fixture(scope="function")
+def engine_new_payloads(test_case: TestCase) -> List[FixtureEngineNewPayload]:
+    """
+    Execution payloads extracted from each block within the test case fixture.
+    Sent to the client under test using the `engine_newPayloadVX` method from the Engine API.
+    """
+    fixture_blocks = [
+        load_dataclass_from_json(FixtureBlock, block.get("rlp_decoded", block))
+        for block in test_case.fixture.blocks
+    ]
+
+    return [
+        FixtureEngineNewPayload.from_fixture_header(
+            globals()[test_case.fixture.fork],
+            block.block_header,
+            block.txs,
+            block.withdrawals,
+            False if block.expected_exception else True,
+            error_code=None,
+        )
+        for block in fixture_blocks
+    ]
```

Comment:
Assuming we have the `ConsumeEngineNewPayload` type in the future, this can simply be one line. :)